### PR TITLE
fix #16238: add wherigo actionbar to most important activities when wherigo is running

### LINKED ISSUE #16238: Wherigo: add "quick access actionbar" to all activities while wherigo is in play
This is an idea which came up in some discussion about Wherigo Player which I want to keep track on.

Idea is to display an "action bar" below the real action bar on all c;:geo activities in case a Wherigo is currently playing. Placement shall be like the "filter bar" in map view/list view, UX view to be discussed. Tapping on this "wherigo bar" would bring the player to the Wherigo activity and/or open any pending dialogs.

### PR DESCRIPTION
**Pull request by eddiemuc**
fix #16238: add wherigo actionbar to most important activities when wherigo is running

image

image

### CODE REVIEW

**Comment by fm-sys** [comment]
I'm wondering whether this implementation should better go into `AbstractNavigationBarActivity.java` / `activity_navigationbar.xml
` instead of the map/list specific layout files?

**Comment by eddiemuc** [comment]
> I'm wondering whether this implementation should better go into `AbstractNavigationBarActivity.java` / `activity_navigationbar.xml
> ` instead of the map/list specific layout files? 

Can be argued. I decided the current way for following reasons:
- as less mixing of wherigo code and "non wherigo" code as possible (team wanted a clean-as-possible solution)
- less complex (a view component can take care of its updates internally while this has to be triggered externally in navigation bar)
- is it ensured 100% that now and in the foreseeable future the set of activities with nav bar and wherigo bar will be the same?

I basically followed the "guide" on how filter-sort-bar was implemented, I just encapsulated the update algo by using a view instead of a layout.

How to proceed here?

**Comment by eddiemuc** [comment]
Updated the list above (basically about the question whether wherigo bar should be shown below or above filter bar if both are present)

**Comment by fm-sys** [comment]
I really like the idea to implement it as view component. This and adding it to `activity_navigationbar.xml` shouldn't conflict each other.

> Update: If we tie wherigo bar to nav bar, then it will be displayed on top of the filter bar in case both is shown (in map and list view). Is this wanted?

I would consider this acceptable or even preferred. The filter bar is tightly coupled with the activity content, so it's better to show it below (thus closer to the activity content). That may be a personal preference of course.

> is it ensured 100% that now and in the foreseeable future the set of activities with nav bar and wherigo bar will be the same?

At least that would be consistent. If implemented as view component it could be added to all bottom navigation activities + a potential white-list of other meanful activities

**Comment by eddiemuc** [comment]
@fm-sys comments are incorporated, CI is green. Let's merge here

File: main/src/main/res/layout/activity_navigationbar.xml
Comment (fm-sys) [inline main/src/main/res/layout/activity_navigationbar.xml:1]: @eddiemuc just noticed while reviewing another PR that you added the whereigo gar to only one of 3 layout file variants. Probably this wasn't intended.

Comment (eddiemuc) [reply main/src/main/res/layout/activity_navigationbar.xml:1]: Indeed...

**Pull request merged**